### DISCUSSION TITLE
TS004F - SmartKnob Payload - ERS-10TZBVK-AA

### DISCUSSION THREAD

**Discussion by alanmgomes** [Koenkk/zigbee2mqtt#12521] [Devices]
I just bought a new smart knob (https://www.aliexpress.com/item/1005003723886310.html?spm=a2g0o.order_list.0.0.1a441802BbPzMC), its my first, so i dont know if its ok, but i found some weird behavior on action report. 
Somehow all actions are reported with a weird payload, like this:

MQTT publish: topic 'zigbee2mqtt/SmartKnob/action', **payload 'hold'**

I think it _should_ be like this:

MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{**"action":"hold"**,"action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":167,"operation_mode":"event","voltage":3000}'

But sometimes I get it this way: 

MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{**"action":""**,"action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":167,"operation_mode":"event","voltage":3000}'

This way, zigbee2mqtt reports an empty string instead of the value.
Does anyone have an idea how to fix this?

![image](https://user-images.githubusercontent.com/42685254/168885475-11b9ea21-65b7-4640-8d56-b1191f5718f1.png)

Here are some logs:
 
Info 2022-05-17 15:24:40MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{"action":"","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":171,"operation_mode":"event","voltage":3000}'
Info 2022-05-17 15:24:40MQTT publish: topic 'zigbee2mqtt/SmartKnob/action', payload 'single'
Info 2022-05-17 15:24:42MQTT publish: topic 'zigbee2mqtt/PlugTomada', payload '{"child_lock":"UNLOCK","current":1.29,"energy":1.72,"indicator_mode":"off/on","linkquality":105,"power":86,"power_outage_memory":"on","state":"ON","update":{"state":"idle"},"update_available":null,"voltage":111}'
Info 2022-05-17 15:24:42MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{"action":"single","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":171,"operation_mode":"event","voltage":3000}'
Info 2022-05-17 15:24:42MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{"action":"","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":171,"operation_mode":"event","voltage":3000}'
Info 2022-05-17 15:24:42MQTT publish: topic 'zigbee2mqtt/SmartKnob/action', payload 'single'
Info 2022-05-17 15:24:46MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{"action":"hold","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":167,"operation_mode":"event","voltage":3000}'
Info 2022-05-17 15:24:46MQTT publish: topic 'zigbee2mqtt/SmartKnob', payload '{"action":"","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"brightness":192,"linkquality":167,"operation_mode":"event","voltage":3000}'
Info 2022-05-17 15:24:46MQTT publish: topic 'zigbee2mqtt/SmartKnob/action', payload 'hold'
Info 2022-05-17 15:24:52

![image](https://user-images.githubusercontent.com/42685254/168886324-305f997e-d39d-40f1-8163-d0d0cdf6f7b0.png)

**Comment by jamferfer** [comment]
Hello @alanmgomes,

I am having exactly the same issue. I had no problem pairing the device but I can't do anything else with it. Every time I perform any action with the switch I get this in the zigbe2mqtt log:
```
Zigbee2MQTT:info  2023-10-31 22:39:34: MQTT publish: topic 'zigbee2mqtt/0x187a3efffefa84b4', payload '{"action":"single","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"linkquality":57,"operation_mode":"event","voltage":3000}'
Zigbee2MQTT:info  2023-10-31 22:39:34: MQTT publish: topic 'zigbee2mqtt/0x187a3efffefa84b4', payload '{"action":"","action_rate":null,"action_step_size":null,"action_transition_time":null,"battery":100,"linkquality":57,"operation_mode":"event","voltage":3000}'
Zigbee2MQTT:info  2023-10-31 22:39:34: MQTT publish: topic 'zigbee2mqtt/0x187a3efffefa84b4/action', payload 'single'
``` 
I have checked in Home Assistant and it seems that with every action performed, the smart knob publishes 2 topics, one with the action performed and another one empty. 

![image](https://github.com/Koenkk/zigbee2mqtt/assets/102029289/e0049bd6-92e9-46b3-bcb1-f32dbe63ea76)

Have you, or anyone found the way to solve this?

**Reply by alanmgomes** [reply]
Hi @jamferfer, yeah it´s a normal beahivour. After any action its is automatic reset, its not a bug, its a feature;
If you go to Homeassistant and check the logs, all the actions are send/capture.
U can make any automations using the states, single, double, etc, and should work.

**Comment by alanmgomes** [comment]
It is a normal behaviour ;)
After any action its is automatic reset, its not a bug, it's a feature;
